client/src/components/History/CurrentHistory/HistoryPanel.vue

```diff
@@ -78,15 +81,15 @@
                     </section>
                     <section v-if="!showAdvanced" class="position-relative flex-grow-1 scroller">
                         <div>
-                            <div v-if="itemsLoaded && itemsLoaded.length == 0 && loading">
+                            <div v-if="loading && !itemsLoaded">
                                 <b-alert class="m-2" variant="info" show>
                                     <LoadingSpan message="Loading History" />
                                 </b-alert>
                             </div>
                             <b-alert v-else-if="isProcessing" class="m-2" variant="info" show>
                                 <LoadingSpan message="Processing operation" />
                             </b-alert>
-                            <div v-else-if="totalItemsInQuery == 0">
+                            <div v-else-if="itemsLoaded.length === 0">
```

Comment:
These two changes need to be revisited to make sure that they do not affect the loading message flow when switching between histories and/or reloading the page.